### ISSUE TITLE
Beefing up m4ri.pc

### ISSUE THREAD

**Issue by kiwifb** [malb/m4ri#2]
I am preparing a new release of BRiAl and it strikes me that the way BRiAl and m4rie access the flags from m4ri is suboptimal. 
Instead of of using AX_M4RI_CFLAGS we should just use the output of of pkg-config. To make it even more interesting I think it should include the library flags for openmp and libpng. Also the optimization flags used don't really belong there. The consumer of m4ri should be able to set its own without worrying about what is in m4ri.pc.
If you think that is a worthwhile idea I'll send a pull request.

**Comment by malb** [comment]
This sounds like a good idea, thanks!

**Comment by kiwifb** [comment]
Getting a better .pc file is very nice but I forgot about the other platforms that may not have pkg-config handy. While pkgconfig will be useful when present we may have to keep AX_M4RI_CFLAGS for when it is not available. Which is slightly annoying.

**Comment by malb** [comment]
True, shall we merge #3 anyway, or is that on hold. Sorry, lost track.

**Comment by kiwifb** [comment]
On hold, there is one more bit I want to add. I have an editor window opened on configure.ac and started to add some stuff... but it is at work and I haven't had time to finish it before leaving.

**Comment by kiwifb** [comment]
I have been a bad boy and completely forgot about finishing. And now I don't remember what exactly I wanted to add. Hum... have to test things on a platform without pkg-config, my OS X laptop is the designed victim.
